packages/kit/src/core/postbuild/analyse.js

```diff
@@ -128,10 +133,21 @@ async function analyse({ manifest_path, env }) {
 				validate_page_exports(page.universal, page.universal_id);
 			}
 
-			prerender = get_option(nodes, 'prerender') ?? false;
+			prerender = get_option(nodes, 'prerender') ?? prerender ?? false;
```

Comment:
I don't think this catches all cases. If the latter of `+page.svelte` has `prerender` set to `true` and `+server.js` has set it to `false` then the final `prerender` value will be `false`, resulting in a false positive. I think we need to save both values independently and then check at the end.